### ISSUE TITLE
GIS importerror vs improperlyconfigured

### ISSUE THREAD

**Issue by strets123** [django-tastypie/django-tastypie#1384]
Using Django 1.7.9 when running tastypie without GeoDjango installed I get an ImpoprtError rather than an ImproperlyConfigured exception, see stacktrace below. This leads to me having to monkey-patch resources.py. Is there something I have wrong in my setup or is this just something that has changed as django has upgraded?

```
$ python manage.py runserver
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/site-packages/django/core/management/__init__.py", line 385, in execute_from_command_line
    utility.execute()
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/site-packages/django/core/management/__init__.py", line 354, in execute
    django.setup()
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/site-packages/django/__init__.py", line 21, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/site-packages/django/apps/registry.py", line 115, in populate
    app_config.ready()
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/site-packages/django/contrib/admin/apps.py", line 22, in ready
    self.module.autodiscover()
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/site-packages/django/contrib/admin/__init__.py", line 23, in autodiscover
    autodiscover_modules('admin', register_to=site)
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/site-packages/django/utils/module_loading.py", line 74, in autodiscover_modules
    import_module('%s.%s' % (app_config.name, module_to_search))
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/home/vagrant/chembiohub_ws/src/cbh_core_ws/cbh_core_ws/admin.py", line 8, in <module>
    from cbh_chembl_ws_extension.projects import ChemregProjectResource
  File "/home/vagrant/chembiohub_ws/src/cbh_chembl_ws_extension/cbh_chembl_ws_extension/projects.py", line 1, in <module>
    from tastypie.resources import ModelResource, ALL, ALL_WITH_RELATIONS
  File "/home/vagrant/chembiohub_ws/src/master/tastypie/resources.py", line 20, in <module>
    from django.contrib.gis.db.models.fields import GeometryField
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/site-packages/django/contrib/gis/db/models/__init__.py", line 8, in <module>
    from django.contrib.gis.db.models.manager import GeoManager  # NOQA
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/site-packages/django/contrib/gis/db/models/manager.py", line 2, in <module>
    from django.contrib.gis.db.models.query import GeoQuerySet
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/site-packages/django/contrib/gis/db/models/query.py", line 6, in <module>
    from django.contrib.gis.db.models.fields import get_srid_info, PointField, LineStringField
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/site-packages/django/contrib/gis/db/models/fields.py", line 4, in <module>
    from django.contrib.gis import forms
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/site-packages/django/contrib/gis/forms/__init__.py", line 2, in <module>
    from .fields import (GeometryField, GeometryCollectionField, PointField,  # NOQA
  File "/home/vagrant/miniconda/bin/chembiohub_ws/lib/python2.7/site-packages/django/contrib/gis/forms/fields.py", line 8, in <module>
    from django.contrib.gis.geos import GEOSException, GEOSGeometry
ImportError: cannot import name GEOSException

```

**Comment by SeanHayes** [comment]
Try removing Django 1.7.9 and installing Django 1.7.10.

We run our core tests against Django 1.7.10 and those don't have Geo Django in INSTALLED_APPS either. I looked through the Django source for 1.7.10 and couldn't find anything that would cause that exception to not get imported; I was even able to import it in a vanilla Python shell.

**Comment by SeanHayes** [comment]
If it still doesn't work, post the output of the following:

To verify your executable is using the right version of Django:
`python manage.py --version`

To test if the ImportError is raised in a shell:
```
python
>>> from django.contrib.gis.geos import GEOSException, GEOSGeometry
```

To make sure the files are present:
`ls -al /path/to/django/contrib/gis/geos`

**Comment by strets123** [comment]
Hi Sean,

Thanks for the quick response to this and the other issue I raise. I have got a sizeable performance improvement with realted resources by upgrading.

The way that an ImportError is forced is by importing the tastypie resources file during the django setup process. This could happen if (as in my case) you use a function from your resources.py in a django admin action function. I have worked around this by using a runtime import inside my function)

It can also happen if tastypie.resources is imported into models.py (which is probably a circular import therefore impossible).

Anyway, if it makes sense we could also catch import error as well as improperlyconfigured. The improperlyconfigured exception is only thrown after django setup has run. See example thread from django leaflet.

https://github.com/makinacorpus/django-leaflet/issues/65
